### PR TITLE
Add types for reload 3.2

### PR DESCRIPTION
**Pull request by Gipphe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/reload/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// Project: https://github.com/alallier/reload#readme
+// Definitions by: Victor Nascimento Bakke <https://github.com/Gipphe>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import express = require('express');
+import ws = require('ws');
+
+declare function reload(
+    app: express.Express,
+    opts: reload.OptionsWaitToStartWebSocketServer,
```
Comment (peterblazejewicz) [inline types/reload/index.d.ts:11]:
```suggestion
    opts?: reload.OptionsWaitToStartWebSocketServer,
```
this one seems to be optional as used in readme:
https://github.com/alallier/reload#express-example
and in source code:
https://github.com/alallier/reload/blob/e7f8234d31a20d0e5142e6959097c1c8011d58c1/lib/reload.js#L9

Comment (Gipphe) [reply types/reload/index.d.ts:11]: It is optional, as noted in the second overload. This overload deals specifically with passing `webSocketServerWaitStart: true` in the `opts` object, which causes `reload` to return a `startWebSocketServer(): Promise<Reload>` method in the resolved object.

**Review by sheetalkamat — APPROVED** [review]

File: types/reload/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/reload/index.d.ts:11]: I'm really sorry, I've missed overloaded func, call me lazy :)

**Review by peterblazejewicz — APPROVED** [review]

**Comment by Gipphe** [comment]
Ready to merge

File: types/reload/index.d.ts (hunk 1 above)
Comment (Gipphe) [reply types/reload/index.d.ts:11]: No worries :) Thank you for reviewing this PR, and thank you for the work you do!

**Pull request merged**